source/Weapon.cpp

```diff
@@ -172,13 +172,23 @@ void Weapon::LoadWeapon(const DataNode &node)
 				damage[HIT_FORCE] = value;
 			else if(key == "piercing")
 				piercing = max(0., min(1., value));
+			else if(key == "damage dropoff")
+			{
+				hasDamageDropoff = true;
+				double maxDropoff = (child.Size() >= 3) ? child.Value(2) : 0.;
+				damageDropoffRange = make_pair(max(0., value), maxDropoff);
+			}
+			else if(key == "damage decay")
+				damageDecay = max(0., value);
 			else
 				child.PrintTrace("Unrecognized weapon attribute: \"" + key + "\":");
 		}
 	}
-	// Sanity check:
+	// Sanity checks:
 	if(burstReload > reload)
 		burstReload = reload;
+	if(damageDropoffRange.first > damageDropoffRange.second)
+		damageDropoffRange.second = Range();
```

Comment:
To make the code more robust/resistant to corner cases I would suggest to use max(Range(), damageDropoffRange.first) here.

Reply:
That doesn't really seem necessary to me since if damageDropoffRange.first > Range(), then damageDropoffRange.second never gets used, as the `distance` of Weapon::DamageDropoff can only be at most equal to the weapon's range, so `if(distance <= minDropoff)` will always be true. (Which is just to say that a minDropoff >= Range is the same as not using this mechanic at all.)